server/controllers/inventory/inventory/core.js

```diff
@@ -278,10 +289,29 @@ async function getItemsMetadata(params) {
   filters.custom('inventory_uuids', 'inventory.uuid IN (?)', params.inventory_uuids);
   filters.setOrder('ORDER BY inventory.code ASC');
 
+  // The query above produces multiple rows for inventory items with more than one tag
+  // (due to the LEFT JOIN).  Therefore use a GROUP BY to reduce to one and add
+  // the tags via the GROUP_CONCAT above.
+  filters.setGroup('GROUP BY inventory.uuid');
+
   const query = filters.applyQuery(sql);
   const parameters = filters.parameters();
   const response = await db.exec(query, parameters);
 
+  // Decode the tags for each inventory item with tags
+  response.forEach(inv => {
+    if (inv.tag_details) {
+      inv.tags = [];
+      const rawTags = inv.tag_details.split('|');
```

Comment:
I wonder if we could chose some sort of unprintable character that users wouldn't have access to - like a UTF8 Byte Order Mark or something.

Reply:
I would prefer not to do that.  It will make it more difficult for anyone doing manual mysql queries.   Non-printable characters can be tricking to handle in terminals.  Second, I personally do not feel that tags should have semicolons or vertical bars.  It seems to me that they should be simple tokens.  I decided not to filter out commas, since I could imagine situations where a comma could be useful in a tag name (and that is why I did not use them for the parsing separators).

Reply:
It is worth noting the French excel uses semicolons instead of commas, so there may be folks who put semicolons in the tags.

Reply:
> It is worth noting the French excel uses semicolons instead of commas, so there may be folks who put semicolons in the tags, just because of language differences.

Good point.  I'll add an Issue about it.  However, picking the separator characters is tricky.